docs/typescript.md

```diff
@@ -0,0 +1,83 @@
+---
+id: typescript
+title: Using TypeScript Tests
+sidebar_label: TypeScript
+---
+
+If you built your automated tests using TypeScript in either Playwright, Puppeteer, TestCafe, or Cypress, you need the following workaround in order to use Testrunner Toolkit.
+
+## What You'll Need
+1. [A Sauce Labs Account](https://saucelabs.com/sign-up)
+2. [TypeScript](https://www.typescriptlang.org/download)
+
+## The Problem
+
+Assuming your `tests` directory setup is similar to the structure below:
+
+__Example Project Setup__
+
+```bash
+.
+└── tests/
+    ├── test.one.spec.ts
+    ├── test.two.spec.ts
+    ├── test.three.spec.ts
+    └── tsconfig.json
+```
+
+Except for the TestCafe image, these `.ts` files cannot run directly on any Testrunner Toolkit images. Therefore in order to make theses test run correclty you must transpile them JavaScript.
+
+## The Solution
+
+1. Ensure you already have `typescript` installed:
+    
+   ```bash
+   npm install -g typescript
+   ```
+
+2. Review your `tsconfig.json` to ensure you've set the `compilerOptions` appropriately.
+ 
+3. Run the TypeScript compiler like so:
+   
+   ```bash
+   npx tsc --project ./tests/tsconfig.json
```

Comment:
Step 1 and 3 are technically incompatible.

If you install Typescript globally:
```
STEP 1: npm install -g typescript
STEP 3: tsc --project ./tests/tsconfig.json
```

If you install locally:
```
STEP 1: npm install typescript
STEP 3: npx tsc --project ./tests/tsconfig.json
```

I would recommend to just show how it's done globally (take out the `npx` from step 3) and forget about documenting how to do it locally. Any savvy Node dev can figure that out on their own.